File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2163.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner ler = new Scanner(System.in);

        int n, m;
        int[][] ter = new int[1024][1024];
        n = ler.nextInt();
        m = ler.nextInt();
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                ter[i][j] = ler.nextInt();
            }
        }
        for(int i=1;i<n-1;i++){
            for(int j=1;j<m-1;j++){
                if(ter[i][j]==42 && ter[i-1][j-1]==7 && ter[i-1][j]==7 && ter[i-1][j+1]==7 && ter[i][j-1]==7 && ter[i][j+1]==7 && ter[i+1][j-1]==7 && ter[i+1][j]==7 && ter[i+1][j+1]==7){
                    System.out.printf("%d %d\n", i+1, j+1);
                    return;
                }
            }
        }
        System.out.printf("0 0\n");
    }
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/2163.lua
n = io.read("*number")
m = io.read("*number")

ter = {}
val = 1
for i=1,n do
    ter[i] = {}
    for j=1,m do
        ter[i][j] = io.read("*number")
    end
end
for i=2,n do
    for j=2,m do
        if(ter[i][j]==42 and ter[i-1][j-1]==7 and ter[i-1][j]==7 and ter[i-1][j+1]==7 and ter[i][j-1]==7 and ter[i][j+1]==7 and ter[i+1][j-1]==7 and ter[i+1][j]==7 and ter[i+1][j+1]==7) then
            print(i.." "..j)
            val = 0
        end
    end
end
if(val==1) then
    print("0 0")
end
